### PR TITLE
src,test: fix V8 engine issues

### PR DESCRIPTION
**Pull request by kfarnung**
* Remove gyp include that no longer exists upstream
* Conditionally include `v8-debug.h`
* Fix test failures for V8 engine
* Remove status entries for missing tests

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

**Comment by kfarnung** [comment]
CI: https://ci.nodejs.org/job/chakracore-test-pull-request/239/

File: test/es-module/es-module.status [hunk 1]
```diff
@@ -23,7 +23,6 @@ test-esm-loader-modulemap : SKIP
 test-esm-main-lookup : SKIP
 test-esm-named-exports : SKIP
 test-esm-namespace : SKIP
-test-esm-ok : SKIP
```
Comment (jackhorton) [inline test/es-module/es-module.status:26]: Was this test just removed?

**Review by jackhorton — COMMENTED** [review]
Looks reasonable, but whats the motivation here?

File: test/es-module/es-module.status (hunk 1 above)
Comment (kfarnung) [reply test/es-module/es-module.status:26]: Looks like it was moved in fddcd6253b237ca68430f44c7614bf52d57c4f97.

**Comment by kfarnung** [comment]
@jackhorton I occasionally like to run a V8 build of the node-chakracore code and make sure the tests all pass.  This ensures that we keep compatibility with V8.  All of these issues (minus the status file cleanup) were preventing the V8 build from passing successfully.

**Comment by kfarnung** [comment]
CI: https://ci.nodejs.org/job/chakracore-test-pull-request/242/

**Comment by kfarnung** [comment]
Ran tests on Windows and Linux:

```
> .\vcbuild.bat v8 test-ci ignore-flaky lint
```

```
$ ./configure --engine=v8
$ make -j4 test
```

No failures

**Review by sethbrenith — APPROVED** [review]
:shipit:

**Review by MSLaguana — APPROVED** [review]

**Comment by kfarnung** [comment]
One more CI before landing, the last one had a bunch of benchmark test timeouts on macOS:  https://ci.nodejs.org/job/chakracore-test-pull-request/243/

**Pull request merged**